server/data/reviews.js

```diff
@@ -89,10 +88,33 @@ const exportedMethods = {
       };
     const insertInfo = await reviewCollection.insertOne(newReview);
     if (insertInfo.insertedCount === 0) throw 'Could not add review';
-    console.log(insertInfo);
-    return "Add review successfully";
-  
+    return insertInfo;
+    },
+
+    async AddReviewToUser(userid,reviewId){
+      const userCollection = await users();
+      const user = await userCollection.findOne({ _id: userid });
+      if (user === null) throw 'No userid with that id';
+      const reviewCollection = await users();
+      let newRest = {
+        _id:reviewId
+      };
+      const updateInfo = await reviewCollection.updateOne(
+        { _id: userid },
+        { $addToSet: { reviews: newRest }} 
+      );
+      if (!updateInfo.matchedCount && !updateInfo.modifiedCount)
+      throw 'Update failed';
+      return "Add review to user successfully!";
     },
+
+    async getUserByReviewId(id){
+      const userCollection = await users();
+      const user = await userCollection.findOne({ 'reviews._id' : id});
+      if (user === null) throw 'No user with that review id';
+      return user;
```

Comment:
user has data like password which we dont want to return, we should only send required fields back like user.name, user.picture, etc. If possible try to filter out roles password etc,